### DISCUSSION TITLE
Per-column search & no results

### DISCUSSION THREAD

**Discussion by bomshteyn** [filamentphp/filament#6349] [General]
When using [Per-column search](https://filamentphp.com/docs/2.x/tables/columns/getting-started#searching-individually) and searching by one of the columns, if it leads to a no results. There is no way to amend your search, only clear all of them. 

<img width="1364" alt="Screenshot 2023-04-28 at 5 16 10 PM" src="https://user-images.githubusercontent.com/4259699/235256004-362b6eaf-08c6-407b-b358-014daa9a47d1.png">

I understand that the reason it happens is that we hide the column names in the table header if there are no results, hence the search boxes disappear with it. I have two questions:

1. why do we need to hide the table header if there are no results, the structure comes from the code not DB?
2. is there a way to override this behavior to force displaying the per-column search if in the scenario of no results

**Comment by danharrin** [comment]
There isn't an advised way to override this behaviour, no. The reason why we hide the table is so that the empty state itself doesn't scroll horizontally when the column headers overflow.

**Comment by avinmaster** [comment]
+1

**Comment by almooradi-dev** [comment]
@bomshteyn @avinmaster 

I did it and I had to issues in the layout, for now :)

Steps:

1. Copy `vendor/filament/tables/resources/views/index.blade.php` to `resources/views/vendor/filament-tables/index.blade.php`
2. Replace `@elseif (($records !== null) && count($records))` by `@elseif (($records !== null))`
3. Add an `else` to the `@if (($records !== null) && count($records))` statement with the code below
```
<tr>
    <td colspan="{{ $columnsCount }}">
        <x-filament-tables::empty-state
            :actions="$getEmptyStateActions()"
            :description="$getEmptyStateDescription()"
            :heading="$getEmptyStateHeading()"
            :icon="$getEmptyStateIcon()"
        />
    </td>
</tr>
```
> You can get this code from the `else` of the `@if (($content || $hasColumnsLayout) && ($records !== null) && count($records))` statement

So the final code will be something like this
```
@if (($records !== null) && count($records))
    ... nothing changed here ...
@else
    <tr>
        <td colspan="{{ $columnsCount }}">
            <x-filament-tables::empty-state
                :actions="$getEmptyStateActions()"
                :description="$getEmptyStateDescription()"
                :heading="$getEmptyStateHeading()"
                :icon="$getEmptyStateIcon()"
            />
        </td>
    </tr>
@endif
```
```
